app/src/main/java/org/jellyfin/androidtv/content/MediaContentProvider.kt

```diff
@@ -0,0 +1,171 @@
+
+import android.app.SearchManager
+import android.content.ContentProvider
+import android.content.ContentValues
+import android.content.Intent
+import android.content.UriMatcher
+import android.database.Cursor
+import android.database.MatrixCursor
+import android.net.Uri
+import android.provider.BaseColumns
+import io.ktor.util.*
+import kotlinx.coroutines.*
+import org.jellyfin.androidtv.TvApp
+import org.jellyfin.androidtv.di.systemApiClient
+import org.jellyfin.apiclient.model.querying.ItemSortBy
+import org.jellyfin.sdk.api.client.KtorClient
+import org.jellyfin.sdk.api.operations.ImageApi
+import org.jellyfin.sdk.api.operations.ItemsApi
+import org.jellyfin.sdk.model.api.*
+import org.jellyfin.sdk.model.serializer.toUUIDOrNull
+import org.koin.core.component.KoinApiExtension
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.inject
+import timber.log.Timber
+import java.text.SimpleDateFormat
+
+
+@KoinApiExtension
+class MediaContentProvider : ContentProvider(), KoinComponent {
+
+	companion object {
+		const val AUTHORITY = "org.jellyfin.androidtv.content"
+		const val SUGGEST_PATH = "suggestions"
+		private const val TICKS_IN_MILLISECOND = 10000
+	}
+
+	private val apiClient by inject<KtorClient>(systemApiClient)
+	private val itemsApi by lazy { ItemsApi(apiClient) }
+	private val imageApi by lazy { ImageApi(apiClient) }
+
+	private val SEARCH_SUGGEST = 1
+
+	private val sURIMatcher = UriMatcher(UriMatcher.NO_MATCH)
+
+	init {
+		sURIMatcher.addURI(AUTHORITY, SUGGEST_PATH + "/" + SearchManager.SUGGEST_URI_PATH_QUERY, SEARCH_SUGGEST)
+		sURIMatcher.addURI(AUTHORITY, SUGGEST_PATH + "/" + SearchManager.SUGGEST_URI_PATH_QUERY + "/*", SEARCH_SUGGEST)
+	}
+
+	override fun onCreate(): Boolean {
+		return true
+	}
+
+	override fun query(uri: Uri, projection: Array<String>?, selection: String?, selectionArgs: Array<String>?,
+					   sortOrder: String?): Cursor? {
+		Timber.d("Query: %s", uri)
+		// Use the UriMatcher to see what kind of query we have and format the db query accordingly
+		when (sURIMatcher.match(uri)) {
+			SEARCH_SUGGEST -> {
+				Timber.d("search suggest: ${selectionArgs?.get(0)} URI: $uri")
+				var limit = 10
+				if (uri.getQueryParameter("limit") != null)
+					limit = uri.getQueryParameter("limit")!!.toInt()
+				return uri.lastPathSegment?.let { getSuggestions(it.toLowerCase(), limit) }
+			}
+			else -> throw IllegalArgumentException("Unknown Uri: $uri")
+		}
+	}
+
+	/**
+	 * Gets the resumable items or returns null
+	 */
+	private suspend fun searchItems(query: String, qLimit: Int): BaseItemDtoQueryResult? {
+		// Get user or return if no user is found (not authenticated)
+		val userId = TvApp.getApplication().currentUser?.id?.toUUIDOrNull() ?: return null
+
+		return itemsApi.getItems(
+				recursive = true,
+				enableTotalRecordCount = true,
+				collapseBoxSetItems = false,
+				excludeLocationTypes = listOf(LocationType.VIRTUAL),
+				sortBy = listOf(ItemSortBy.DateCreated),
```

Comment:
```suggestion
				sortBy = listOf("DateCreated"),
```

`ItemSortBy` is from the old apiclient, the SDK doesn't have an enum or constant for it right now.

Reply:
You could also use `ItemFields.DATE_CREATED.serialName` or `ItemFields.DATE_CREATED.toString()`